packages/api/src/providers/tenovi.ts

```diff
@@ -78,24 +112,42 @@ export class Tenovi extends Provider {
    * Disconnects the device from the user's Tenovi Gateway.
    * Optionally updates the user's connected devices list.
    *
-   * @param connectedUser The user to disconnect the device from
-   * @param deviceId      The device to disconnect
-   * @param updateUser    Whether to update the user's connected devices list
+   * @param connectedUser The user to disconnect the device from.
+   * @param deviceId      The device to disconnect.
+   * @param updateUser    Whether to update the user's connected devices list.
+   * @param extraParams   The extra request parameters.
+   * @param rawParams     The raw request parameters.
    */
   async disconnectDevice(
     connectedUser: ConnectedUser,
     deviceId: string,
-    updateUser = true
+    updateUser: boolean,
+    extraParams?: TenoviExtraParams,
+    rawParams?: RawParams
   ): Promise<void> {
     const connectedDevices = connectedUser.providerMap?.tenovi?.connectedDeviceIds;
+    let xTenoviApiKey = extraParams?.xTenoviApiKey;
+    if (!xTenoviApiKey && rawParams) {
+      xTenoviApiKey = getTenoviHeaderOrFail(tenoviApiKeyPropName, rawParams);
+    }
+    if (!xTenoviApiKey) throw new BadRequestError(`Missing ${tenoviApiKeyPropName} header.`);
+
+    let xTenoviClientName = extraParams?.xTenoviClientName;
+    if (!xTenoviClientName && rawParams) {
+      xTenoviClientName = getTenoviHeaderOrFail(tenoviClientNamePropName, rawParams);
+    }
+    if (!xTenoviClientName) throw new BadRequestError(`Missing ${tenoviApiKeyPropName} header.`);
+
     if (connectedDevices && connectedDevices.includes(deviceId)) {
-      const url = `${Tenovi.URL}/${Tenovi.API_PATH}/hwi/unlink-gateway/${deviceId}/`;
+      const url = `${Tenovi.URL}/${Tenovi.API_PATH}/${encodeURIComponent(
```

Comment:
Wouldn't it be better to encode the name when we first parse it from the request? Then any usage, including logs and reports to Sentry, its safe.